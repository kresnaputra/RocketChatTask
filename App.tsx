import StoryBook from './storybook';
export default StoryBook;
// import App from './src';
// export default App;
