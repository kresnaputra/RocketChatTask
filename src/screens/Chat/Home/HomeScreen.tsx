import React from 'react';
import ChatListItem from '../../../components/ChatListItem';
import {FlatList, View, StyleSheet} from 'react-native';
import {colors} from '../../../constans/colors';
import {ChatNavProps} from '../../../types/navigation/chat';

const Dummy = [
  {
    id: 1,
    name: 'Jhon',
    chatDescription: 'Hello, I already take the computer',
    time: '18:00',
    countChat: 3,
    isRead: false,
  },
  {
    id: 2,
    name: 'Devi',
    chatDescription: 'Hey nice to meet you',
    time: '17:00',
    countChat: 0,
    isRead: true,
  },
];

const HomeScreen = ({navigation}: ChatNavProps<'Home'>) => {
  const detailNavigation = (id: number) => {
    navigation.navigate('Detail');
  };

  return (
    <FlatList
      style={styles.container}
      renderItem={({item}) => (
        <ChatListItem {...item} onPress={detailNavigation} />
      )}
      data={Dummy}
      keyExtractor={(item) => `${item.id}`}
      ItemSeparatorComponent={() => <View style={styles.line} />}
      ListHeaderComponent={() => <View style={{marginVertical: 5}} />}
      ListFooterComponent={() => <View style={{marginVertical: 5}} />}
    />
  );
};

const styles = StyleSheet.create({
  line: {
    marginVertical: 5,
    height: 1,
    marginLeft: 60,
    backgroundColor: colors.gray,
  },
  container: {flex: 1, backgroundColor: 'white', paddingHorizontal: 10},
});

export default HomeScreen;
