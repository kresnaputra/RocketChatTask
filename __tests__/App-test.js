/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';
import HomeScreen from '../src/screens/Chat/Home/HomeScreen';

const navigation = {
  navigate: jest.fn(),
};

jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest.fn().mockImplementation((_, reducers) => reducers),
  };
});

it('renders correctly', async () => {
  await act(async () => {
    renderer.create(<App />);
  });
});

const TreeHomeScreen = <HomeScreen navigation={navigation} />;

it('HomeScreen UI', () => {
  expect(TreeHomeScreen).toMatchSnapshot();
});

