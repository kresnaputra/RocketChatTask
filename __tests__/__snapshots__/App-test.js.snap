// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen UI 1`] = `
<Memo(Connect(HomeScreen))
  navigation={
    Object {
      "navigate": [MockFunction],
    }
  }
/>
`;
